kura/org.eclipse.kura.rest.inventory.provider/src/main/java/org/eclipse/kura/internal/rest/inventory/InventoryRestService.java

```diff
@@ -0,0 +1,287 @@
+        }
+    }
+
+    /**
+     * POST method.
+     *
+     * Start selected container.
+     *
+     */
+    @POST
+    @RolesAllowed("inventory")
+    @Path("/containers/_start")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response startContainer(final String bundleJson) {
+        try {
+            return makeInventoryDoExecRequest(buildKuraMessage(InventoryHandlerV1.START_CONTAINER, bundleJson));
+        } catch (KuraException e) {
+            throw DefaultExceptionHandler.toWebApplicationException(e);
+        }
+    }
+
+    /**
+     * POST method.
+     *
+     * Stop selected container.
+     *
+     */
+    @POST
+    @RolesAllowed("inventory")
+    @Path("/containers/_stop")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response stopContainer(final String bundleJson) {
+        try {
+            return makeInventoryDoExecRequest(buildKuraMessage(InventoryHandlerV1.STOP_CONTAINER, bundleJson));
+        } catch (KuraException e) {
+            throw DefaultExceptionHandler.toWebApplicationException(e);
+        }
+    }
+
+    /**
+     * GET method.
+     *
+     * Lists all the available container images.
+     *
+     */
+    @GET
+    @RolesAllowed("inventory")
+    @Path("/images")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getImages() {
+        try {
+            return makeInventoryDoGetRequest(
+                    buildKuraMessage(Arrays.asList(InventoryHandlerV1.RESOURCE_CONTAINER_IMAGES), ""));
+        } catch (KuraException e) {
+            throw DefaultExceptionHandler.toWebApplicationException(e);
+        }
+    }
+
+    /**
+     * POST method.
+     *
+     * Delete selected container image.
+     *
+     */
+    @POST
+    @RolesAllowed("inventory")
+    @Path("/images/_delete")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response deleteImage(final String bundleJson) {
+        try {
+            return makeInventoryDoExecRequest(buildKuraMessage(InventoryHandlerV1.DELETE_IMAGE, bundleJson));
+        } catch (KuraException e) {
+            throw DefaultExceptionHandler.toWebApplicationException(e);
+        }
+    }
+
+    private KuraMessage buildKuraMessage(List<String> requestObject, String body) {
+
+        Map<String, Object> payloadProperties = new HashMap<>();
+        payloadProperties.put(ARGS_KEY.value(), requestObject);
+
+        KuraPayload kuraPayload = new KuraPayload();
+        kuraPayload.setBody(body.getBytes());
+
+        return new KuraMessage(kuraPayload, payloadProperties);
+    }
+
+    private Response makeInventoryDoGetRequest(KuraMessage kuraMessage) throws KuraException {
+        KuraMessage inventoryResponse = inventoryHandlerV1.doGet(null, kuraMessage);
+        return Response.ok(inventoryResponse.getPayload().getBody()).build();
```

Comment:
This method should also handle the case when the response payload contains a failure status. For example at [1] setting the HTTP response code accordingly.

[1] https://github.com/eclipse/kura/blob/dae19acb7902ab32c72b1b9ae6e2d96a8d96dd8e/kura/org.eclipse.kura.core.inventory/src/main/java/org/eclipse/kura/core/inventory/InventoryHandlerV1.java#L464